platform/linux-generic/pktio/pcap.c

```diff
@@ -276,9 +278,32 @@ static int pcapif_recv_pkt(pktio_entry_t *pktio_entry, int index ODP_UNUSED,
 			break;
 		}
 
-		packet_parse_layer(pkt_hdr,
-				   pktio_entry->s.config.parser.layer,
-				   pktio_entry->s.in_chksums);
+		if (pktio_cls_enabled(pktio_entry)) {
+			odp_packet_t new_pkt;
+
+			ret = cls_classify_packet(pktio_entry, data,
+						  pkt_len, pkt_len,
+						  &new_pool, pkt_hdr, true);
+			if (ret) {
+				odp_packet_free(pkt);
+				continue;
+			}
+			if (new_pool != odp_packet_pool(pkt)) {
```

Comment:
No need to call `odp_packet_pool ()` function, you can use `pcap->pool` directly.

Reply:
Ack